### ISSUE TITLE
volumes in WorkflowTemplates are not carried over to Workflows

### ISSUE THREAD

**Issue by foobarbecue** [argoproj/argo#3341]
Checklist:

 * [x] I've included the version.
 * [x] I've included reproduction steps.
 * [x] I've included the workflow YAML.
 * [ ] I've included the logs.
 
**What happened**:

This Workflow suceeds, and writes a file test.txt to the host:

```
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  generateName: voltest-
  namespace: default
spec:
  entrypoint: test-vols
  volumes:
    - name: nac-volume
      hostPath:
        path: /c/testdir
        type: DirectoryOrCreate

  templates:
    - name: test-vols
      container:
        image: ubuntu
        volumeMounts:
          - name: nac-volume
            mountPath: /data
        command: [sh, -c]
        args: ["echo test > /data/test.txt"]
```

However, if you convert it to a WorkflowTemplate and submit a Workflow based on it, then it fails with 
`volume 'nac-volume' not found in workflow spec`

Here's the WorkflowTemplate version:

```
apiVersion: argoproj.io/v1alpha1
kind: WorkflowTemplate
metadata:
  name: voltesttmpl
  generateName: voltest-
  namespace: default
spec:
  entrypoint: test-vols
  volumes:
    - name: nac-volume
      hostPath:
        path: /c/testdir
        type: DirectoryOrCreate

  templates:
    - name: test-vols
      container:
        image: ubuntu
        volumeMounts:
          - name: nac-volume
            mountPath: /data
        command: [sh, -c]
        args: ["echo test > /data/test.txt"]
```

**What you expected to happen**:

I expected the `volumes` to be carried over to the Template from the WorkflowTemplate.

**How to reproduce it (as minimally and precisely as possible)**:

See "what happened" above.

**Environment**:

- Argo version:
```
$ argo version
argo: v2.9.0-rc4
  BuildDate: 2020-06-26T21:54:40Z
  GitCommit: 5b109bcb9257653ecbae46e6315c8d65842de58a
  GitTreeState: clean
  GitTag: v2.9.0-rc4
  GoVersion: go1.13.4
  Compiler: gc
  Platform: linux/amd64
```
- Kubernetes version :
```
$ kubectl version -o yaml
clientVersion:
  buildDate: "2019-11-13T11:23:11Z"
  compiler: gc
  gitCommit: b3cbbae08ec52a7fc73d334838e18d17e8512749
  gitTreeState: clean
  gitVersion: v1.16.3
  goVersion: go1.12.12
  major: "1"
  minor: "16"
  platform: linux/amd64
serverVersion:
  buildDate: "2020-01-15T08:18:29Z"
  compiler: gc
  gitCommit: e7f962ba86f4ce7033828210ca3556393c377bcc
  gitTreeState: clean
  gitVersion: v1.16.6-beta.0
  goVersion: go1.13.5
  major: "1"
  minor: 16+
  platform: linux/amd64
```

**Message from the maintainers**:

If you are impacted by this bug please add a 👍 reaction to this issue! We often sort issues this way to know what to prioritize.

**Comment by alexec** [comment]
@foobarbecue do you know if this is a new problem?

**Comment by foobarbecue** [comment]
@alexec I don't know -- this was my first time trying volumes with WorkflowTemplates. So it's probably not new. BTW I also tried https://github.com/argoproj/argo/blob/master/examples/volumes-emptydir.yaml as a WorkflowTemplate with the same result.

**Comment by foobarbecue** [comment]
I'm working on this now:
- Adding the missing properties from WorkflowSpec to WorkflowTemplateSpec in ui/src/models
- Correcting the getWorkflow function in ui/src/app/workflow-templates/components/workflow-template-details/workflow-template-details.tsx so that it copies those properties over

Presumably there needs to be a server-side fix as well, for when you do `argo submit --from workflowtemplate/mywft`

I am more and more convinced that argo shot itself in the foot when it created workflowtemplates instead of adding features to templates :-p

**Comment by alexec** [comment]
@sarabala1979 would you like to own this please?

**Comment by foobarbecue** [comment]
Yep, can do. I expect to have a PR today.

**Comment by foobarbecue** [comment]
Oh, sorry I was working on an old fork... there is no more WorkflowTemplateSpec in ui/src/models, it just references TemplateSpec... so that's a good thing. So it's just a matter of fixing the getWorkflow function now (and then the CLI fixes).

**Comment by foobarbecue** [comment]
Draft PR: https://github.com/argoproj/argo/pull/3441

**Comment by alexec** [comment]
I this #3436 fixes this too.

**Comment by foobarbecue** [comment]
Ah, cool, I'll try out #3437 and close this if it works

**Comment by foobarbecue** [comment]
I just tried it, at 81cba832ed1d4f5b116dc9e43f1f3ad79c190c44 and get 
volume 'nac-volume' not found in workflow spec

**Comment by alexec** [comment]
@foobarbecue - this needs two fixes. One to remove the panic that obscures the error (this was a pre-existing bug) an another to fix the actual bug #3441.

**Comment by foobarbecue** [comment]
Ah, ok. When we make a WF from a WFT, and the WF is referring to the WFT using workflowTemplateRef, do we want the volumes and stuff to stay on the WFT? Or do we want to copy them to the WF? My #3441 copies them to the WF.

**Comment by alexec** [comment]
Fixed in v2.9

**Comment by foobarbecue** [comment]
Just to confirm, I'm using this now and it works well!
